internal/engine/integration/procedure_test.go

```diff
@@ -160,6 +160,60 @@ func Test_Procedures(t *testing.T) {
 				{"satoshi", "hello world"},
 			},
 		},
+		{
+			name: "string functions",
+			procedure: `procedure string_funcs() public view {
+				$val := 'hello world';
+				$val := $val || '!!!';
+				$val := upper($val);
+				if $val != 'HELLO WORLD!!!' {
+					error('upper failed');
+				}
+				$val := lower($val);
+				if $val != 'hello world!!!' {
+					error('lower failed');
+				}
+
+				if bit_length($val) != 112 {
+					error('bit_length failed');
+				}
+				if char_length($val) != 14 or character_length($val) != 14 or length($val) != 14 {
+					error('length failed');
+				}
+				if octet_length($val) != 14 {
+					error('octet_length failed');
+				}
+				$val := rtrim($val, '!');
+				if $val != 'hello world' {
+					error('rtrim failed');
+				}
+				$val := ltrim($val, 'h');
+				if $val != 'ello world' {
+					error('ltrim failed');
+				}
+				$val := lpad($val, 11, 'h');
+				if $val != 'hello world' {
+					error('lpad failed');
+				}
+				$val := rpad($val, 12, '!');
+				if $val != 'hello world!' {
+					error('rpad failed');
+				}
+				if overlay($val, 'xx', 2, 5) != 'hxxworld!' {
+					error('overlay failed');
+				}
+				if position('world', $val) != 7 {
+					error('position failed');
+				}
+				if substring($val, 7, 5) != 'world' {
```

Comment:
maybe also add test for different args? functions like `ltrim`,`overlay`,`substring`(and more) support different args num

Reply:
ok done